### PR TITLE
feat: add sentry logging

### LINKED ISSUE #11: SENTRY_DSN does not work with serverless lambda :(

### PR DESCRIPTION
**Pull request by jakebolam**
Fixes #11

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,7 +51,6 @@ module.exports.serverless = appFn => {
     // Do the thing
     console.log(`Received event ${e}${event.body.action ? ('.' + event.body.action) : ''}`)
     if (event) {
-      try {
         await probot.receive({
```
Comment (jakebolam) [inline index.js]: Let errors boil up so sentry logs them

File: index.js [hunk 2]
```diff
@@ -62,20 +62,8 @@ module.exports.serverless = appFn => {
...
-          statusCode: 500,
-          body: JSON.stringify(err)
-        })
-      }
     } else {
-      console.error({ event, context })
-      context.done(null, 'unknown error')
+      throw new Error(`Unknown Error, No event ${context}`)
     }
-    return context.done(null, {
```
Comment (jakebolam) [inline index.js]: un used code path

File: index.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ const loadProbot = appFn => {
   probot = probot || createProbot({
     id: process.env.APP_ID,
     secret: process.env.WEBHOOK_SECRET,
+    webhookProxy: process.env.WEBHOOK_PROXY_URL,
```
Comment (jakebolam) [inline index.js]: Support the smee proxy. (Should we actually port all the config from probot-run ?) https://github.com/probot/probot/blob/master/bin/probot-run.js#L24-L31

**Comment by jakebolam** [comment]
@JasonEtco added sentry support

**Comment by jakebolam** [comment]
Bump @JasonEtco any chance we can get this in?

**Comment by JasonEtco** [comment]
No need to bump @jakebolam - its only been a few days since you've opened this. I'm not sure that this library should include the Sentry plugin by default - its more of a size concern than anything else. @probot/maintainers feel free to weigh in!

In the meantime, you can load the plugin yourself:

```js
module.exports = app => {
  if (process.env.SENTRY_DSN) {
    require('probot/lib/apps/sentry')(app)
  }

  // Your regular app code...
  app.on('*', () => {})
}
```

**Comment by JasonEtco** [comment]
Going to close this out, given that there's [a workaround above](https://github.com/probot/serverless-lambda/pull/13#issuecomment-455827957). And it seems like @gr2m and I agree that this doesn't need to be included in the extension. Thanks for starting the conversation @jakebolam ❤️

**Comment by tcbyrd** [comment]
I've been playing around with [Lambda's layers](https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html) and I think that's the route I'd like to support for things like Sentry. There are some existing solutions that have started implementing things this way like IO Pipe<sup>[1]</sup> and Datadog<sup>[2]</sup>. I know other FaaS providers are working on similar solutions, so that will probably be the "right" way to do this so the function can remain as small as possible.

[1]: https://read.iopipe.com/iopipe-for-aws-lambda-byol-1a184ae5a5
[2]: https://www.datadoghq.com/blog/datadog-lambda-layer/

**Comment by jakebolam** [comment]
no worries, sounds good to me.

**Pull request closed without merging**